### ISSUE TITLE
Extend Safety Phase with a new check

### ISSUE THREAD

**Issue by magnus-madsen** [flix/flix#2126]
Add a check to ensure that lattice variables do not flow to relations (but the other way is allowed) i.e.:

```
Lat(123, x) :- Rel(x). // OK
Rel(x) :- Lat(123, x). // NOT OK
```

I think this can be done by maintaining a map from variable symbols to their type (relational/lattice) and reporting an error whenever a variable is used as both.

**Comment by magnus-madsen** [comment]
@JonathanStarup I think this is something you can do! We can chat more about the details.

**Comment by magnus-madsen** [comment]
@JonathanStarup Did we fix this already?

**Comment by JonathanStarup** [comment]
This was fixed by #3624
